### ISSUE TITLE
Import failed: Slot number must increase

### ISSUE THREAD

**Issue by EclesioMeloJunior** [ChainSafe/gossamer#2636]
## Describe the bug

- While running a gossamer node with two other substrate nodes I notice that the substrate throws the following error:
```
2022-06-30 14:32:51 💔 Error importing block 0xf919e19068653baf95680af89d1751ce43cf4cc9422c0fa8e98300f470a430fc: consensus error: Import failed: Slot number must increase: parent slot: 414153490, this slot: 414153490
```

This block was created by the gossamer node:
```
2022-06-30T14:32:42-04:00 INFO built block 155 with hash 0xf919e19068653baf95680af89d1751ce43cf4cc9422c0fa8e98300f470a430fc, state root 0x8305859db24d16d1a34cee766dbf7d5b3132f3d4d0c57d31f21ec9e765752d68, epoch 6 and slot 414153490	babe.go:L541	pkg=babe
```

The slot number specified in the error messages was claimed by the gossamer node as a secondary vrf slot in the epoch 6:
```
2022-06-30T14:32:40-04:00 DBUG claimed secondary slot, for slot number: 414153490	crypto.go:L110	pkg=babe
2022-06-30T14:32:40-04:00 DBUG epoch 6: claimed secondary vrf slot 414153490	epoch.go:L239	pkg=babe
```

**Comment by kishansagathiya** [comment]
I don't think this is related to secondary slot claiming. I have seen this happen for primary block as well. This has more to do with some time related inaccuracies. This error occurs only for the first slot in the epoch.
If I just change https://github.com/ChainSafe/gossamer/blob/3a471d91f61c1ba10022ff11403b9cebb71c8666/lib/babe/epoch_handler.go#L88 to 
```
		if authoringSlot <= currSlot {
```

we would not see the problem anymore. But that would mean that first slot would never get used, so that's bad.

**Comment by kishansagathiya** [comment]
Different error `block has an unknown parent`, fun!!
```
2022-07-18 17:19:07 💔 Error importing block 0xaf9a9621dd46c50b668967ccc78f983aad789127658143b780296121e34c0f97: block has an unknown parent    
```

**Comment by kishansagathiya** [comment]
> @kishansagathiya I think this is not the reason, the error `genesis hash mismatch` does not enable us to sync with the peers, could you compare the block 0 hash in the gossamer and in the peer they should be the same otherwise the block will not be imported. Maybe the genesis spec you're using is different from the one is used in the peer

If you see `genesis hash mismatch`, doesn't necessarily mean that genesis are mismatching

**Comment by EclesioMeloJunior** [comment]
🤔 hm, does this error you've reported here -> https://github.com/ChainSafe/gossamer/issues/2636#issuecomment-1187577614 happened at the beginning of the sync when both peers start a connection? Or the error happened when peers were connected for some time exchanging blocks?

**Comment by kishansagathiya** [comment]
So, what I am noticing here is that substrate start a new epoch with the proposal of a particular block at epoch start slot. Let's say this is the xth block. Gossamer is able to import this block. But then gossamer also build a block at the same slot, but it builds (x+1)th block. 

I believe, gossamer should check the slot in the blocks that it imports. If gossamer believes that it is it's time to author a block, it should ignore the received block and instead of authoring x+1th block, it should author xth block.

**Comment by kishansagathiya** [comment]
I was think about what can cause the above scenario.
- inconsistency in slot calculation, like may be rust code and go code running simultaneously, it does not produce the same slot number. Otherwise, why would I have a block for a slot number that is ongoing?
- Also, should I even import a block for the ongoing slot number? I feel like no.
